loopdb/store.go

```diff
@@ -265,8 +268,18 @@ func deserializeUpdates(swapBucket *bbolt.Bucket) ([]*LoopEvent, error) {
 	// Deserialize and collect each swap update into our slice of swap
 	// events.
 	var updates []*LoopEvent
-	err := stateBucket.ForEach(func(_, v []byte) error {
-		event, err := deserializeLoopEvent(v)
+	err := stateBucket.ForEach(func(k, v []byte) error {
+		if v != nil {
+			return fmt.Errorf("expected state sub-bucket for %x", k)
+		}
+
+		updateBucket := stateBucket.Bucket(k)
```

Comment:
Could also remove the `v!=nil` and check `updateBucket!=nil` instead, slightly more intuitive?

Reply:
good suggestion, fixed